apm-sniffer/apm-sdk-plugin/pom.xml

```diff
@@ -47,6 +47,7 @@
         <module>sharding-sphere-3.x-plugin</module>
         <module>sharding-sphere-4.x-plugin</module>
         <module>sharding-sphere-4.x-RC3-plugin</module>
+        <module>sharding-sphere-4.1.0-plugin</module>
```

Comment:
Could you merge these 4 as one module? The source codes are duplicated logically.

Reply:
Each version of the package has a different path, so different versions need to be provided

Reply:
That is not my asking. I am talking on the maven module perspective.

Reply:
Maybe they could merge into a module,like this:

sharding-sphere-plugin
       ---- sharding-sphere-3.x-plugin
       ---- sharding-sphere-4.x-plugin
       ---- sharding-sphere-4.x-RC3-plugin
       ---- sharding-sphere-4.1.0-plugin       

What do you think, maybe there will be more submodules in the future.

Reply:
Skywalking doesn't need the different plugin jars to identify the versions, your plugin instrumentation witness definition makes that. So, once you moved the codes into one plugin, you could use separate packages for different versions. Others wouldn't be loaded.

Reply:
```
sharding-sphere-plugin
---- sharding-sphere-3.x-plugin
---- sharding-sphere-4.x-plugin
---- sharding-sphere-4.x-RC3-plugin
---- sharding-sphere-4.1.0-plugin
```

If you want, you could, but my point is, there are so many ShardingSphere plugin jars. Are they really so different?

Reply:
sharding-sphere, In previous versions, the named paths of packages have changed, so have many jar, but after 4.0 ~5.0 , it to share one.   5.0 maybe provide annother jar.

Reply:
Hi Sheng,
Yes, the class name and method name are so different almost every ShardingSphere release.
As I know, 4.0.0 is different from 4.0.0-RC3, and 4.1.0 is differnent from 4.0.1.
For the 4.1.1 under releasing, same as 4.1.0, they can share a same plugin.

Reply:
I still hope you could move all codes in one jar. You could see in the current codebase, there is no plugin for other libs like this :)

Reply:
Anyway, this is purely my hope.

Reply:
Hi，Sheng,
I see your point, But ShardingSphere Changing the package path at release time is too frequent at 4.0.0-RC ,  But inevitably, there will be path changes during version updates, like springMVC3,4,5  They can't be combined into one.  It's just my idea...